### PR TITLE
Support ABFSS and Hadoop Logical Relation in Column Level Lineage

### PR DESCRIPTION
**Pull request by wjohnson**
### Problem

Currently, the Column Lineage Input Field Collectors work mainly for Spark SQL operations and Data Source V2.

This leaves out normal dataframe operations like inserting into HDFS without the use of a Hive table.

Column Lineage should support this scenario as many users will want to see column lineage for operations outside of SQL and Hive Metastore backed tables.

Closes: #963

### Solution

I introduce a `extractDatasetIdentifier(HadoopFsRelation relation)` which uses similar logic to `InsertIntoHadoopFsRelationVisitor` to pull out the path on the HDFS compliant file system. This was tested on ABFSS and DBFS (Databricks FileSystem) to prove that lineage could be extracted using non-sql commands.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

### CODE REVIEW

**Review by pawel-big-lebowski — COMMENTED** [review]
Looks like some test is failing. 
Beside that, it looks great 👍 💯

**Comment by wjohnson** [comment]
@pawel-big-lebowski test is updated and passing!  Thank you for your review!

File: integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/InputFieldsCollector.java [hunk 1]
```diff
@@ -104,4 +112,25 @@ private static List<DatasetIdentifier> extractDatasetIdentifier(CatalogTable cat
...
+    List<DatasetIdentifier> inputDatasets = new ArrayList<DatasetIdentifier>();
+    List<Path> paths =
+        JavaConversions.asJavaCollection(relation.location().rootPaths()).stream()
+            .collect(Collectors.toList());
+
+    for (Path p : paths) {
+      String namespace = PlanUtils.namespaceUri(p.toUri());
+      inputDatasets.add(new DatasetIdentifier(p.toUri().getPath(), namespace));
+    }
+    if (inputDatasets.isEmpty()) {
```
Comment (pawel-big-lebowski) [inline integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/InputFieldsCollector.java:130]: Shouldn't `return inputDatasets;` be just enough?

Comment (wjohnson) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/InputFieldsCollector.java]: 🤦yeah... that's a good one.  Thank you for catching and improving, Pawel.

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**